app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityUtils.kt

```diff
@@ -63,3 +74,52 @@ fun DeleteActivityAlertDialog(
       },
       dismissButton = { TextButton(onClick = { onDismissRequest() }) { Text("Cancel") } })
 }
+/**
+ * A dialog for filtering activities by type. Displays a list of available activity types with
+ * checkboxes to toggle filters.
+ *
+ * @param selectedFilters A set of currently selected [ActivityType] filters.
+ * @param onFilterChanged Callback invoked when a filter checkbox is checked or unchecked.
+ * @param onDismiss Callback invoked to close the dialog.
+ */
+@Composable
+fun FilterDialog(
+    selectedFilters: Set<ActivityType>,
+    onFilterChanged: (ActivityType, Boolean) -> Unit,
+    onDismiss: () -> Unit
+) {
+  AlertDialog(
+      onDismissRequest = { onDismiss() },
+      modifier = Modifier.testTag("filterActivityAlertDialog"),
+      title = {
+        Text(
+            text = stringResource(R.string.filter_activity_type),
+            style = MaterialTheme.typography.titleMedium)
+      },
+      text = {
+        Column(
+            modifier = Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(10.dp)) {
```

Comment:
To me the spacing between rows is quite big, wdyt of making it a bit smaller?

Reply:
You're right :) It looks better with smaller spacing! I'll make the change.